osCam/pathbuilder/views.py

```diff
@@ -0,0 +1,284 @@
+			next_path_list = getNextPathList(curUser=aUser, pathBuilder=default_path_builder)
+			# make sure to turn off after editing..
+			storage_rule.update_or_create(
+				update=False
+			)
+			return render(request, 'core/home.html', {
+				"update_path_dialog":True,
+				"path":default_path_builder.path, 
+				"nextpaths":next_path_list
+				})
+
+	else:
+		if Path.objects.filter(user=aUser).exists():
+			display = Path.objects.get(user=aUser)
+			
+			# display_storage_rule = storage_rule.objects.get(user=aUser)
+			display_nextPathList = getNextPathList(aUser, display)
+			return render(request, 'core/home.html', {
+					"update_path_dialog":True,
+					"path":display.path,
+					"nextpaths":display_nextPathList
+					})
+		else:
+			# display Default $Path info
+			# display_storage_rule = StorageHandler.objects.get(user=aUser)
+			return render(request, 'core/home.html', {
+					"update_path_dialog":False,
+					"path":PathDataDialog.DEFAULT_PATH,
+					"nextpaths":[PathDataDialog.SENTINAL_OPEN_PATH]
+					})
+@login_required
+def home(request):
+	# path = None
+	# cpath="/"
+	pathDataDialog = PathDataDialog()
+	this_user = User.objects.get(id=request.user.id)
+	path_list = []
+	print("REQUEST_post: ",request.POST)
+
+	print("REQUEST_get: ",request.GET)
+
+	if request.method == 'POST':
+		if "$PATH" in request.POST:
+			print("%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%")
+			builder = request.POST.get("$PATH")
+			print("\n\n%%%[MAIN__HOME.path-builder]: {}".format(builder))
+			print("%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%")
+			ui_context = displaySetStoragePath(request)
+			return render(request, 'core/home.html', ui_context)
+		else:
+			print("[MAIN_HOME.request.isEMpty()]")
+			
+	elif request.method == 'GET':
+		print("\n\nFound Path: {}".format(str(request.GET)))
+		print("&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&")
+		print("Request.GET")
+		print("&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&")
+		displayCurrentStorageView(request, this_user)
+			
+		next_path_list: list = []
+		default_path_builder: Path = None
+		storage_rule: StorageHandler = None
+		if "$EDIT" in request.GET:
+			print("\n\n$edit REQUEST: {}".format(request.GET))
+			if Path.objects.get(user=this_user).exists():
+				default_path_builder = Path.objects.get(user=this_user)
+				# print(f"Most Recent Path: {default_path_builder.path} \n\n")
+				default_path_builder.storage.objects.update_or_Create(
+					update=True
+				)
+				next_path_list = getNextPathList(curUser=this_user, pathBuilder=default_path_builder)
+				context = {"pathBuilder":default_path_builder, "nextPathList": next_path_list}
+				print("path-builder: {}".format(default_path_builder))
+				print("\n\tNextPath: {}".format(next_path_list))
+				storage_rule=default_path_builder.storage.objects.get(id=default_path_builder.id)
+				next_path_list = getNextPathList(curUser=this_user, pathBuilder=default_path_builder)
+				# make sure to turn off after editing..
```

Comment:
just checking, it says here to make sure to turn something off after editing? did you remember to turn it off?

Reply:
ya when the user is done with the form, I was just noting to myself that I needed to make sure the path were only queried when the PathForm was in use. Also, I didn't mean to push this view with this PR so I'm temporarily deleting this page.

Reply:
This page does not seem to be removed from the PR?